contribs/src/main/java/com/netflix/conductor/contribs/http/LongRunningHttpTask.java

```diff
@@ -0,0 +1,392 @@
+/**
+ * Copyright 2016 Netflix, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ *
+ */
+package com.netflix.conductor.contribs.http;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.netflix.conductor.auth.AuthManager;
+import com.netflix.conductor.auth.ForeignAuthManager;
+import com.netflix.conductor.common.metadata.tasks.Task;
+import com.netflix.conductor.common.metadata.tasks.Task.Status;
+import com.netflix.conductor.common.run.Workflow;
+import com.netflix.conductor.core.DNSLookup;
+import com.netflix.conductor.core.config.Configuration;
+import com.netflix.conductor.core.events.ScriptEvaluator;
+import com.netflix.conductor.core.execution.WorkflowExecutor;
+import com.netflix.conductor.service.MetricService;
+import datadog.trace.api.Trace;
+import org.apache.commons.lang3.StringUtils;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static java.util.Objects.isNull;
+
+@Singleton
+public class LongRunningHttpTask extends GenericHttpTask {
```

Comment:
Would it be better to extend the HTTP Task and decorate the start method with the update thread so that we can keep the HTTP functionality in one place..

Reply:
Instead of extending the http task we can pass a new param in the request input for ex:  long_runnning_http =true and based on that if its true we run the logic to update unack timeout.Otherwise it will run as normal http task